hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -56,6 +56,11 @@ class MergeOnReadSnapshotRelation(sqlContext: SQLContext,
   protected val mergeType: String = optParams.getOrElse(DataSourceReadOptions.REALTIME_MERGE.key,
     DataSourceReadOptions.REALTIME_MERGE.defaultValue)
 
+  override def imbueConfigs(sqlContext: SQLContext): Unit = {
+    super.imbueConfigs(sqlContext)
+    sqlContext.sparkSession.sessionState.conf.setConfString("spark.sql.parquet.enableVectorizedReader", "true")
```

Comment:
If we read the incremental and snapshot tables at the same time, the current configuration may result in inaccurate incremental data.
Because after enabling Vectorized Reader, parquet's recordLevelFilter will not work, and incremental query depends on recordLevelFilter to work, which may cause incremental query to generate a lot of irrelevant data.

Can we provide a configuration to decide whether to enable Vectorized Reader?

Reply:
I agree, we should make this option configurable.

Reply:
Or better, just disable vec reader when query type is incremental.